firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -373,6 +367,22 @@ private void addReadTime() {
     }
   }
 
+  private void addReadTimeMigrationWatermark() {
+    if (!tableContainsColumn("target_globals", "first_document_without_read_time")) {
+      db.execSQL("ALTER TABLE target_globals ADD COLUMN first_document_without_read_time TEXT");
+    }
+
+    SQLiteStatement insertFirstMissingReadTime =
+        db.compileStatement(
+            "UPDATE target_globals SET first_document_without_read_time = ("
+                + "SELECT path FROM remote_documents "
+                + "WHERE read_time_seconds IS NULL ORDER BY path LIMIT 1)");
```

Comment:
Per my comment elsewhere, we could probably simplify and just store "" initially.